psydac/fem/projectors.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+
+from psydac.linalg.kron     import KroneckerDenseMatrix
+from psydac.core.bsplines   import hrefinement_matrix
+from psydac.linalg.identity import IdentityStencilMatrix
+from psydac.linalg.stencil  import StencilVectorSpace
+
+def knots_to_insert(coarse_grid, fine_grid, tol=1e-14):
+#    assert len(coarse_grid)*2-2 == len(fine_grid)-1
+    intersection = coarse_grid[(np.abs(fine_grid[:,None] - coarse_grid) < tol).any(0)]
+    assert abs(intersection-coarse_grid).max()<tol
+    T = fine_grid[~(np.abs(coarse_grid[:,None] - fine_grid) < tol).any(0)]
```

Comment:
I think we could store a boolean array instead of computing it twice:
```python
indices = (np.abs(fine_grid[:,None] - coarse_grid) < tol).any(0)
```
By the way, why do you add a dimension to the `fine_grid` array? Here it looks like `fine_grid` is 1D, while `coarse_grid` is 2D... Why should that be?

Reply:
it's not computed twice, it's a different one in each time, but  it's now much clearer in  [86f53a3](https://github.com/pyccel/psydac/pull/241/commits/86f53a3e46503da7dac62c58b2edeedc4f631418)

Reply:
I wanted to compute the difference between each point in the fine grid with every point in the coarse grid, that's why I made it into a 2D array and  with the broadcasting rule, it will compute what I wanted.

Reply:
I see, you are computing `N_fine * N_coarse` distances in 1D.